lib/ansible/modules/packaging/os/snap.py

```diff
@@ -0,0 +1,212 @@
+'''
+
+EXAMPLES = '''
+# Install "foo" snap
+- name: Install foo
+  snap:
+    name: foo
+
+# Remove "foo" snap
+- name: Remove foo
+  snap:
+    name: foo
+    state: absent
+
+# Install a snap with classic confinement
+- name: Install "foo" with option --classic
+  snap:
+    name: foo
+    classic: yes
+'''
+
+RETURN = '''
+msg:
+    description: An informative message on the task's output
+    type: str
+    returned: Always
+classic:
+    description: Whether or not the snaps were installed with the classic confinement
+    type: bool
+    returned: When snaps are installed
+changed:
+    description: True if one or more snaps are installed/removed
+    type: bool
+    returned: on success
+err:
+    description: The stderr output
+    type: str
+    returned: on failure
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def clean_err(err):
+    # Remove whitespaces and newlines that the snap CLI outputs.
+    return err.replace('\n', '').replace('      ', '').replace('"', '\'')
+
+
+def snap_exists(module, snap_name):
+    snap_path = module.get_bin_path("snap", True)
+    cmd_parts = [snap_path, 'info', snap_name]
+    cmd = ' '.join(cmd_parts)
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    return rc == 0
+
+
+def is_snap_installed(module, snap_name):
+    snap_path = module.get_bin_path("snap", True)
+    cmd_parts = [snap_path, 'list', snap_name]
+    cmd = ' '.join(cmd_parts)
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    return rc == 0
+
+
+def install_snaps(module, snap_names):
+    snaps_not_installed = list()
+    for snap_name in snap_names:
+        if is_snap_installed(module, snap_name):
+            continue
+        snaps_not_installed.append(snap_name)
+
+    if not snaps_not_installed:
+        snaps_already_installed = ', '.join(snap_names)
+        module.exit_json(msg="Snap(s) already installed: %s" % snaps_already_installed, classic=module.params['classic'], changed=False)
+
+    # Transform the list into a string with whitespace-separated snaps
+    snaps_to_install = ' '.join(snaps_not_installed)
+    # Create a string with commas for the output
+    snaps_installed = ', '.join(snaps_not_installed)
+
+    if module.check_mode:
+        module.exit_json(msg="Snap(s) that would have been installed: %s" % snaps_installed, classic=module.params['classic'], changed=True)
+
+    classic = '--classic' if module.params['classic'] else ''
+
+    snap_path = module.get_bin_path("snap", True)
+    cmd_parts = [snap_path, 'install', snaps_to_install, classic]
+    cmd = ' '.join(cmd_parts)
+
+    # Actually install the snaps
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    if rc == 0:
+        module.exit_json(msg="Snap(s) installed: %s" % snaps_installed, classic=module.params['classic'], changed=True)
+    else:
+        err = clean_err(err)
+        module.fail_json(msg="Something went wrong.", classic=module.params['classic'], err=err)
```

Comment:
wire here `stderr=err` (original), without cleanup. also `stdout=out`

Reply:
and maybe `rc=rc` would be useful as well

Reply:
Is rc useful? I mean if it's not 0, the module fails anyway.